### PR TITLE
Check for IE8 and IE9 and fix CORS

### PR DESCRIPTION
**Pull request by rnisthal**
Check for IE8 and IE9 and fix CORS for those browsers.

### CODE REVIEW

File: layer/vector/KML.js [hunk 1]
```diff
@@ -17,16 +17,34 @@ L.KML = L.FeatureGroup.extend({
...
+		            var xml = new ActiveXObject('Microsoft.XMLDOM');
+		            xml.loadXML(xdr.responseText);
+		            cb(xml, options);
+		        }
+		    }
+		    setTimeout(function () { xdr.send(); }, 0);
+		} else {
+		    req.open('GET', url, async);
+		    try {
+		        req.overrideMimeType('text/xml'); // unsupported by IE
```
Comment (brunob) [inline layer/vector/KML.js]: Are you sure we need this test after your changes ?

**Comment by rnisthal** [comment]
What test are you refering to?
the if (xdr.responseText) ?

Well, it is redundant, xdr.responseText has all the server response, which is then turned into an xml dom.
There should be some error handling there to test that it is a valid XML though.

**Comment by brunob** [comment]
My inline comment was on line 39 test, https://github.com/rnisthal/leaflet-plugins/blob/master/layer/vector/KML.js#L39 which point ```// unsupported by IE```

**Comment by rnisthal** [comment]
Ok :D
I'm new with github
IE10 will hit that area, because it does support CORS
I don't know if it supports overrideMimeType so I didn't change any of that code

**Comment by brunob** [comment]
jshint is crying on this merge, can you fix it please ?

**Comment by brunob** [comment]
@rnisthal i've merged and fixed many jshints warning in cd7002096db7536fe94123a156bb5f50831292e6 ; could you test it and confirm that it works well ?

**Comment by brunob** [comment]
Just for the record, some links about this bug : 

http://perrymitchell.net/article/xdomainrequest-cors-ie9/
http://docs.telerik.com/kendo-ui/framework/datasource/cors
https://github.com/MoonScript/jQuery-ajaxTransport-XDomainRequest/blob/master/jQuery.XDomainRequest.js

**Comment by rnisthal** [comment]
@brunob I tested it on my site using IE8 and it's working :)

**Comment by brunob** [comment]
Merged, thanks for the feedback.

**Pull request closed without merging**